src/Microsoft.OData.Edm/Schema/EdmTargetPath.cs

```diff
@@ -0,0 +1,94 @@
+//---------------------------------------------------------------------
+// <copyright file="EdmTargetPath.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.OData.Edm
+{
+    /// <inheritdoc/>
+    public class EdmTargetPath : IEdmTargetPath
+    {
+        private List<IEdmElement> segments;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EdmTargetPath"/> class.
+        /// </summary>
+        /// <param name="segments">Target path segments.</param>
+        public EdmTargetPath(IEnumerable<IEdmElement> segments)
+        {
+            EdmUtil.CheckArgumentNull(segments, nameof(segments));
+
+            this.segments = new List<IEdmElement>(segments);

```

Comment:
>new List<IEdmElement>(segments); [](http://example.com/codeflow?start=28&length=32)

It should be better to call "ToList()" than "use new List<T>", right?

ToList() checkes IEnumerable<T>, if it's a IList<T>, it just returns, there's no new List<T> created.